src/libumockdev-preload.c

```diff
@@ -1431,35 +1431,44 @@ int statx(int dirfd, const char *pathname, int flags, unsigned mask, struct stat
     return r;
 }
 
+static bool is_fd_in_mock(int fd, const char *subdir)
+{
+    static char fdpath[PATH_MAX];
+    static char linkpath[PATH_MAX];
+    snprintf(fdpath, sizeof fdpath, "/proc/self/fd/%i", fd);
+    int orig_errno = errno;
+    ssize_t linklen = readlink(fdpath, linkpath, sizeof linkpath);
```

Comment:
No need to use our own overridden `readlink` version here.

Reply:
Well spotted, thanks! Fixed.